modules/light-clients/08-wasm/types/upgrade_test.go

```diff
@@ -98,536 +106,137 @@ func (suite *TypesTestSuite) TestVerifyUpgradeGrandpa() {
+					return nil, 0, contractError
+				})
+			},
+			contractError,
+		},
+	}
 
-// 			revisionNumber := clienttypes.ParseChainID(tmClientState.ChainId)
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			// reset suite
+			suite.SetupWasmWithMockVM()
 
-// 			newChainID, err = clienttypes.SetRevisionNumber(tmClientState.ChainId, revisionNumber+1)
-// 			suite.Require().NoError(err)
+			endpoint = wasmtesting.NewWasmEndpoint(suite.chainA)
+			err := endpoint.CreateClient()
+			suite.Require().NoError(err)
 
-// 			var tmUpgradedClient exported.ClientState
-// 			tmUpgradedClient = ibctm.NewClientState(newChainID, ibctm.DefaultTrustLevel, trustingPeriod, ubdPeriod+trustingPeriod, maxClockDrift, clienttypes.NewHeight(revisionNumber+1, clientState.GetLatestHeight().GetRevisionHeight()+1), commitmenttypes.GetSDKSpecs(), upgradePath)
-// 			tmUpgradedClient = tmUpgradedClient.ZeroCustomFields()
-// 			tmUpgradedClientBz, err := clienttypes.MarshalClientState(suite.chainA.App.AppCodec(), tmUpgradedClient)
-// 			suite.Require().NoError(err)
+			clientState := endpoint.GetClientState().(*types.ClientState)
 
-// 			upgradedClient = types.NewClientState(tmUpgradedClientBz, clientState.CodeHash, clienttypes.NewHeight(revisionNumber+1, clientState.GetLatestHeight().GetRevisionHeight()+1))
-// 			upgradedClientBz, err = clienttypes.MarshalClientState(suite.chainA.App.AppCodec(), upgradedClient)
-// 			suite.Require().NoError(err)
+			upgradedClient = types.NewClientState([]byte("client state"), clientState.CodeHash, clienttypes.NewHeight(0, clientState.GetLatestHeight().GetRevisionHeight()+1))
+			upgradedConsState = &types.ConsensusState{[]byte("consensus state")}
 
-// 			var tmUpgradedConsState exported.ConsensusState //nolint:gosimple // ignore error for test
-// 			tmUpgradedConsState = &ibctm.ConsensusState{
-// 				Timestamp:          time.Now(),
-// 				Root:               commitmenttypes.NewMerkleRoot([]byte("root hash")),
-// 				NextValidatorsHash: []byte("01234567890123456789012345678901"),
-// 			}
-// 			tmUpgradedConsStateBz, err := clienttypes.MarshalConsensusState(suite.chainA.App.AppCodec(), tmUpgradedConsState)
-// 			suite.Require().NoError(err)
+			tc.malleate()
 
-// 			upgradedConsState = &types.ConsensusState{
-// 				Data: tmUpgradedConsStateBz,
-// 			}
-// 			upgradedConsStateBz, err = clienttypes.MarshalConsensusState(suite.chainA.App.AppCodec(), upgradedConsState)
-// 			suite.Require().NoError(err)
+			proofUpgradedClient = []byte("proof upgraded client")
+			proofUpgradedConsState = []byte("proof upgraded consensus state")
 
-// 			tc.setup()
+			err = clientState.VerifyUpgradeAndUpdateState(
+				suite.ctx,
+				suite.chainA.Codec,
+				suite.store,
```

Comment:
should we be referencing the updated context/store here? Or it it the intention to use the static values set by setup?

Reply:
Intention is to use static values, other tests go the same route. Pretty sure this should play no role in the mocking case, correct? All we basically test for here is that the store we pass in eventually does get written to.